src/features/unified-share-modal/ContactRestrictionNotice.js

```diff
@@ -0,0 +1,139 @@
+// @flow
+
+import * as React from 'react';
+import noop from 'lodash/noop';
+import getProp from 'lodash/get';
+import uniqueId from 'lodash/uniqueId';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import type { InjectIntlProvidedProps } from 'react-intl';
+
+import FormattedCompMessage from '../../components/i18n/FormattedCompMessage';
+import Param from '../../components/i18n/Param';
+import Label from '../../components/label';
+import PlainButton from '../../components/plain-button';
+import InlineNotice from '../../components/inline-notice';
+import LoadingIndicator from '../../components/loading-indicator/LoadingIndicator';
+import SingleSelectField from '../../components/select-field/SingleSelectField';
+
+import ContactsEmailsTooltip from './ContactsEmailsTooltip';
+import messages from './messages';
+
+import type { SelectOptionProp } from '../../components/select-field/props';
+import type { contactType as Contact } from './flowTypes';
+
+import './ContactRestrictionNotice.scss';
+
+type Props = {
+    error?: React.Node,
+    isLoading?: boolean,
+    justificationReasons: Array<SelectOptionProp>,
+    onRemoveRestrictedExternalContacts: () => void,
+    onSelectJustificationReason: (justificationReasonOption: SelectOptionProp) => void,
+    restrictedExternalEmails: Array<string>,
+    selectedContacts: Array<Contact>,
+    selectedJustificationReason: ?SelectOptionProp,
+} & InjectIntlProvidedProps;
+
+const ContactRestrictionNotice = ({
+    error,
+    intl,
+    isLoading,
+    justificationReasons,
+    onRemoveRestrictedExternalContacts,
+    onSelectJustificationReason,
+    restrictedExternalEmails,
+    selectedContacts,
+    selectedJustificationReason,
+}: Props) => {
+    const compMessageId = uniqueId('compMessage');
+    const restrictedExternalContacts = selectedContacts.filter(({ value }) => restrictedExternalEmails.includes(value));
+    const restrictedExternalContactCount = restrictedExternalContacts.length;
+
+    if (!restrictedExternalContactCount) {
+        return null;
+    }
+
+    const RemoveButton = ({ children }: { children: React.Node }) => (
+        <PlainButton
+            className="bdl-ContactRestrictionNotice-removeBtn"
+            data-resin-target="removeBtn"
+            onClick={onRemoveRestrictedExternalContacts}
+        >
+            {children}
+        </PlainButton>
+    );
+
+    // TODO:
+    // Switch to react-intl v3+ api once migration is complete. FormattedCompMessage is now
+    // deprecated and has some issues with components nested within <Plural/>, hence why we
+    // use two messages and use unique keys as a workaround.
+
+    const noticeDescriptionSingular = (
```

Comment:
Unfortunately ran into some issues with this deprecated component, which we'll have to work around since there's currently no viable alternative. Note that `<Plural />` requires you to specify the complete message for both cases, so the standard component usage is almost as verbose as the workaround.